### PR TITLE
fix: reindex mongo error when datamodel change

### PR DESCRIPTION
**Pull request by AlvaroVega**
Related with PR https://github.com/telefonicaid/fiware-cygnus/pull/2211

https://www.mongodb.com/docs/manual/reference/method/db.collection.createIndex/

### CODE REVIEW

**Comment by AlvaroVega** [comment]
Maybe IndexKeySpecsConflict happens when DM is changed and previous index exists with another key.
Make sense @fgalan @Madhu-NEC

**Comment by AlvaroVega** [comment]
maybe related https://github.com/telefonicaid/fiware-cygnus/issues/2291 ?

**Comment by fgalan** [comment]
> Maybe IndexKeySpecsConflict happens when DM is changed and previous index exists with another key. Make sense @fgalan @Madhu-NEC ?

It seems the code was already prepared for re-indexing. I understand that it was not working until you have added the IndexKeySpecsConflict exception to the catch clause.

Maybe in the past (i.e. old versions of MongoDB or the driver) the exception was IndexOptionsConflict so it was working that time but not now? In this case, I think the strategy of adding the new exception (IndexKeySpecsConflict) is a good idea.

**Comment by AlvaroVega** [comment]
> > Maybe IndexKeySpecsConflict happens when DM is changed and previous index exists with another key. Make sense @fgalan @Madhu-NEC ?
> 
> It seems the code was already prepared for re-indexing. I understand that it was not working until you have added the IndexKeySpecsConflict exception to the catch clause.
> 
> Maybe in the past (i.e. old versions of MongoDB or the driver) the exception was IndexOptionsConflict so it was working that time but not now? In this case, I think the strategy of adding the new exception (IndexKeySpecsConflict) is a good idea.

IMHO:
IndexOptionsConflict -> when the same index (name and keys) already exits
IndexKeySpecsConflict -> when an index with the same name but different keys (i.e. DM was changed) already exist.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 1]
```diff
@@ -215,7 +215,8 @@ public void createIndex(MongoDatabase db, String collectionName, BasicDBObject k
         try {
             db.getCollection(collectionName).createIndex(keys, options);
         } catch(Exception e) {
-            if (e.getMessage().contains("IndexOptionsConflict")) {
+            if (e.getMessage().contains("IndexOptionsConflict") ||
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:218]:
```suggestion
            // Our guess is:
            // IndexOptionsConflict -> when the same index (name and keys) already exits
            // IndexKeySpecsConflict -> when an index with the same name but different keys (i.e. DM was changed) already exist
            if (e.getMessage().contains("IndexOptionsConflict") ||
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**